sources/engine/Stride.Graphics/Vulkan/PipelineState.Vulkan.cs

```diff
@@ -229,8 +229,8 @@ private unsafe void CreateRenderPass(PipelineStateDescription pipelineStateDescr
             if (hasDepthStencilAttachment)
                 attachmentCount++;
 
-            var attachments = new VkAttachmentDescription[attachmentCount];
-            var colorAttachmentReferences = new VkAttachmentReference[renderTargetCount];
+            var attachments = new AttachmentDescription[attachmentCount];
```

Comment:
One optimization you could consider in general (potentially for the future) is to conditionally stackalloc:
```cs
Span<AttachmentDescription> attachments = (attachmentCount * sizeof(AttachmentDescription)) < 512 ? stackalloc AttachmentDescription[attachmentCount] : new AttachmentDescription[attachmentCount];
```

Might be a bit verbose for this codebase, but it's just a little thing to not invoke the GC if you don't need it. The `512` is so we don't risk stack overflows and don't allocate too much on the stack - `512` is just a number I've pulled from my backside and I usually scale it according to how much code like this I have i.e. if I have one of these "conditional stackallocs" then I use 512, if I have two I use 256, etc...